### DISCUSSION TITLE
Handling error with typescript problem

### DISCUSSION THREAD

**Discussion by oskar-gmerek** [TanStack/query#3436] [Q&A] [answered]
Hello, 

I'm learning so please do not beat me ;)

`{newUserMutation.isError && <p>{newUserMutation.error?.message}</p>}`

That line of code is working as expected, but when I trying to build a project, then I getting that error:

`Type error: Property 'message' does not exist on type 'unknown'.`

Is there anyone who will be able help me to sort it out?

Kind Regards,
Oskar

**Comment by TkDodo** [comment, marked as answer]
in JavaScript, you can throw anything, it doesn't have to be an Error. If you `throw 5`, error would be a number. In TypeScript, [since v4.4](https://devblogs.microsoft.com/typescript/announcing-typescript-4-4/#use-unknown-catch-variables), if you use try/catch, the error you get in the catch clause will also be of type `unknown`. So you need to narrow it first, e.g. with:

```
{newUserMutation.error instanceOf Error && <p>{newUserMutation.error?.message}</p>}
```

I am covering this in my blog here: https://tkdodo.eu/blog/react-query-and-type-script#what-about-error

**Reply by oskar-gmerek** [reply]
Thank you so much. I was looked at your blog before writing that question, but missed that part.

Btw. Great blog!

**Reply by dindustack** [reply]
Thanks, this solved my issue
